### DISCUSSION TITLE
[Suggestion] Custom Palettes for Software Rendering Emulation

### DISCUSSION THREAD

**Discussion by Immorpher** [andrei-drexler/ironwail#23] [Ideas]
Depending how tricky it is to implement, having custom palettes might be a cool feature on the software rendering emulation side. It might be particularly useful for modern maps which use colored lightmaps.

You could swap in Doom, Duke Nukem, or Quake 2 palettes.

You could make use of the full bright color space to add a few more shades for modern maps.

I am happy to make a few palettes for experimentation.

**Comment by andrei-drexler** [comment]
Normally, I'd say this is outside the scope of this project, since IMO it would be turning Quake into something it's not (with the original palette being an important part of Quake's DNA). On the other hand, though, this is the only port that does real-time palettization, so I guess it could be interesting as a 'what if' experiment.

**Reply by Immorpher** [reply]
Indeed, it wouldn't be for the classic aesthetic so it's a bit outside of the scope, but it's a curiousity. I am particularly interested in Piotr's generated palettes here: https://pixelation.org/index.php?topic=18752.0

Since Quake lends on the darker side, I probably would do a non-linear brightness adjustment on those palettes to favor the darker shades a bit more. :)
